Library/Homebrew/service.rb

```diff
@@ -371,7 +391,7 @@ def to_plist
       base = {
         Label:            @formula.plist_name,
         ProgramArguments: command,
-        RunAtLoad:        @run_type == RUN_TYPE_IMMEDIATE,
+        RunAtLoad:        @run_type == RUN_TYPE_IMMEDIATE || (@run_at_load == true && @run_type == RUN_TYPE_INTERVAL),
```

Comment:
Please extract this into a separate method, it's a bit gnarly having it inside a `Hash`